### PR TITLE
Feat: integrate flow

### PR DESCRIPTION
**Pull request by bhargav175**
**What**:
A feature PR for jntegrating **flow-type** with the website.

**Why**:

Fix issue #65. 
To leverage static type checking with flow.

**How**:

- Added .flowconfig as well as a flow-typed directory for existing packages.
- Haven't changed any of the .js files yet. This PR only caters to handling flow integration.
- .eslintignore flow-typed directories to prevent rewrite with prettier-eslint as the flow-typed files are auto-generated.

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[ignore]
+<PROJECT_ROOT>/.*/__tests__/.*
+<PROJECT_ROOT>/docs/.*
```
Comment (kentcdodds) [inline .flowconfig]: We don't have a `/docs/` directory.

**Comment by bhargav175** [comment]
@kentcdodds  I forgot to include the `flow` as npm script. When you run flow, it exits without errors. However, when you start adding the flow annotation to the js files, it works as expected, throws errors as expected. I think the config went alright.

File: .flowconfig (hunk 1 above)
Comment (bhargav175) [inline .flowconfig]: Oh. Right. Let me fix that.

**Comment by bhargav175** [comment]
@kentcdodds Haven't actually added flow annotations to the js files yet because that would be a different feature request( which would mean a fix on potentially every js file in there). However, flow annotations do work if they are added to any of the js files. This PR only covers the setup for flow. Is that good enough ?

**Comment by kentcdodds** [comment]
Yep! That's great! Thank you!

**Comment by kentcdodds** [comment]
If you'd like to add flow to a file or two that would be cool!

**Comment by bhargav175** [comment]
Alright. Will do that.

**Pull request merged**